### PR TITLE
add supabase client

### PR DESCRIPTION
**Pull request by timkpaine**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by timkpaine** [comment]
@anand2312 @dreinon @J0 @leynier anybody want to be a maintainer here?

**Comment by J0** [comment]
Hey @timkpaine,

Thanks for putting this PR together! Would be great to have conda support - feel free to add me as a maintainer. Let me also ping the rest of the maintainers to see if they want to hop onboard.

Joel

**Comment by timkpaine** [comment]
https://github.com/orgs/conda-forge/teams/help-python ok now actually all set

File: recipes/supabase/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - python >=3.7
+    - postgrest >=0.10.2,<0.11.0
+    - realtime >=0.0.5
+    - gotrue >=0.5.0
+    - httpx >=0.23.0
+    - storage3 >=0.3.5
+    - supafunc >=0.2.2
+    - python-semantic-release
+
+test:
```
Comment (xhochy) [inline recipes/supabase/meta.yaml:32]: Can you add a pip check here?

File: recipes/realtime/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - pip
+    - poetry
+  run:
+    - python >=3.7
+    - dataclasses >=0.6,<0.7
+    - python-dateutil >=2.8.1
+    - typing-extensions >=4.2.0
+    - websockets >=10.3
+
+test:
```
Comment (xhochy) [inline recipes/realtime/meta.yaml:29]: Can you add a pip check here?

File: recipes/postgrest/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{% set name = "postgrest" %}
```
Comment (xhochy) [inline recipes/postgrest/meta.yaml:1]: We should mirror the PyPI name here, too.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/supabase/meta.yaml (hunk 1 above)
Comment (timkpaine) [reply recipes/supabase/meta.yaml:32]: No it fails, see comments.

File: recipes/realtime/meta.yaml (hunk 2 above)
Comment (timkpaine) [reply recipes/realtime/meta.yaml:29]: No it fails, see comments.

File: recipes/postgrest/meta.yaml (hunk 3 above)
Comment (timkpaine) [reply recipes/postgrest/meta.yaml:1]: postgrest is the correct name, postgrest pypi package is dead and conda convention is to omit "-python" when this happens (e.g. https://github.com/conda-forge/perspective-feedstock is "perspective-python" on pypi, but perspective on conda due to stale package)

**Comment by timkpaine** [comment]
> @conda-forge/help-python all set, had to disable some `pip check` due to some upstream dep issues which im looking into

@xhochy note comment about `pip check`. the results of `pip check` are wrong for `realtime` (I validated this manually) so the check is omitted, then it is again omitted for `supabase` since `realtime` is a dep.

**Comment by xhochy** [comment]
I would normally request a comment on why it has been omitted but with a merged upstream fix, I'm fine.

**Comment by timkpaine** [comment]
> I would normally request a comment on why it has been omitted but with a merged upstream fix, I'm fine.

if you have `dataclasses` as an unscoped dep `pip check` will fail. You need something like `dataclasses; python<3.7` for it to pass.

**Pull request merged**